src/main/java/frc/robot/hardware/signal/InputSignal.java

```diff
@@ -0,0 +1,70 @@
+package frc.robot.hardware.signal;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import org.littletonrobotics.junction.LogTable;
+import org.littletonrobotics.junction.inputs.LoggableInputs;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.DoubleFunction;
+
+
+public abstract class InputSignal<T> implements LoggableInputs {
+
+	public abstract static class DoubleSignal extends InputSignal<Double> {
+
+		protected DoubleSignal() {
+			super(0.0);
+		}
+
+	}
+
+	public abstract static class AngleSignal extends InputSignal<Rotation2d> {
+
+		private final DoubleFunction<Rotation2d> toAngle;
+
+		protected AngleSignal(DoubleFunction<Rotation2d> toAngle) {
+			super(new Rotation2d());
+			this.toAngle = toAngle;
+		}
+
+		protected void setNewValues(double[] newValues) {
+			setNewValues(Arrays.stream(newValues).mapToObj(toAngle).toArray(Rotation2d[]::new));
+		}
+
+		protected void setNewValues(double newValue) {
+			setNewValues(toAngle.apply(newValue));
+		}
```

Comment:
If i have something that returns a rotation 2d, I won't need to use the DoubleFunction and the functions `setNewValues`
. But according to  your code I will need to provide a toAngle even though i won't use it

this can be fixed with a default constructor that doesn't require a function